doc/performance.md

```diff
@@ -1,39 +1,257 @@
 # Performance
 
-There is a Vagrantfile with performance inspection tools like SystemTap.
+- [**General guidelines**](#general-guidelines)
+- [**Default caching**](#default-caching)
+- [**Async reporting threads**](#async-reporting-threads)
+- [**Batching policy**](#batching-policy)
 
-```shell
-vagrant up
-vagrant ssh
-```
+This document gives some general guidelines to debug performance issues in
+APIcast. It also introduces the caching modes available and explains how they
+can help in increasing performance.
 
-APIcast is mounted into `/home/vagrant/app` so you can start it by:
+## General guidelines
 
-```shell
-cd app
-bin/apicast
-```
+When experiencing performance issues in APIcast, the first thing that we
+recommend to do is to try to determine the component that is responsible for
+those issues. In a typical APIcast deployment, there are three components to
+take into account:
 
-For profiling with stapxx it is recommended to start just one process and worker:
+- APIcast
+- The 3scale backend that authorizes requests and keeps track of the usage
+- The upstream API
 
-```shell
-bin/apicast -c examples/configuration/echo.json  > /dev/null
-```
+In order to measure the latency that APIcast plus the 3scale backend introduce,
+the first thing that we need to do is to measure the latency of the upstream
+API. There are many tools that can be used for that:
+[wrk](https://github.com/wg/wrk), [vegeta](https://github.com/tsenart/vegeta),
+[JMeter](https://jmeter.apache.org/), etc. The next step consists of using the
+same tool to run the same benchmark, but this time, pointing to APIcast instead
+of pointing to the upstream API directly. Comparing those results will give us
+an idea of the latency introduced by APIcast and the 3scale backend.
+
+If you see that the latency introduced by APIcast and the 3scale backend is too
+high, and you are using a self-managed APIcast that authorizes against the
+3scale backend SaaS, the first thing you can do is to make a simple request to
+the 3scale backend from the same machine where APIcast is deployed and measure
+the latency. Backend exposes a simple endpoint that just returns the version:
+https://su1.3scale.net/status. An authorization call requires more work because
+it needs to verify some keys, limits, and enqueue background jobs, among other
```

Comment:
```suggestion
verifying some keys and limits, enqueuing background jobs, as well as ...
```
_among other things_ looks like a filler. If there are more tasks, specify them. Otherwise, you can finish the sentence.

Reply:
My intention was to mention only the features that are relevant for APIcast, and at the same time, don't give the impression that those are the only things that backend does.

Reply:
Accuracy is paramount in technical documentation. We don't want to leave gaps that might cause confusion. 
I understand that this section refers mainly to APIcast. Better if you mention that these features are relevant for APIcast. Regarding the backend's functionality, you can mention a couple of tasks, or link to another doc for further reference.